src/main/java/seedu/duke/storage/FavStorage.java

```diff
@@ -47,24 +51,38 @@ public void readFile() throws FileNotFoundException {
      * @param entry if number of delimiter not 1 or command does not contain "/"
      */
     private void readLine(String entry) {
-        String[] entryWords = entry.split("\\|");
-        if (entryWords.length != 2) {
+        String[] entryWords = entry.split("\\|",2);
+        if (entryWords.length != 2 || entryWords[1].isBlank()) {
             isCorrupted = true;
             return;
         }
-        if (!entryWords[0].contains("/")) {
+        if (!isValidCommand(entryWords[0])) {
             isCorrupted = true;
             return;
         }
         FavList.addFav(new Fav(entryWords[0], entryWords[1]));
 
     }
 
+    /**
+     * Checks if command type is a valid command.
```

Comment:
This comment might be confusing. Could specify that it is a command that can be favourited.